src/main/java/com/ordestiny/tdd/designpattern/factorypattern/AnimalFactory.java

```diff
@@ -0,0 +1,19 @@
+package com.ordestiny.tdd.designpattern.factorypattern;
+
+import com.ordestiny.tdd.designpattern.factorypattern.animal.Animal;
+import com.ordestiny.tdd.designpattern.factorypattern.animal.AnimalEnum;
+
+import java.security.InvalidParameterException;
+import java.text.MessageFormat;
+
+/**
+ * @author Rachit Parikh
+ */
+public class AnimalFactory {
+    public static Animal getAnimal(AnimalEnum animalEnum) {
+        if(animalEnum.getInstance() == null){
+            throw new InvalidParameterException();
+        }
+        return animalEnum.getInstance();

```

Comment:
You should put your type creation logic in this Factory class. Otherwise, this factory is useless.

Reply:
Type creation logic?